parquet-column/src/main/java/org/apache/parquet/column/values/dictionary/DictionaryValuesWriter.java

```diff
@@ -577,7 +613,8 @@ public void writeFloat(float v) {
     public DictionaryPage createDictionaryPage() {
       if (lastUsedDictionarySize > 0) {
         // return a dictionary only if we actually used it
-        PlainValuesWriter dictionaryEncoder = new PlainValuesWriter(lastUsedDictionaryByteSize, maxDictionaryByteSize);
+        PlainValuesWriter dictionaryEncoder = new PlainValuesWriter(lastUsedDictionaryByteSize, maxDictionaryByteSize, allocator);
+        dicionaryWriters.add(dictionaryEncoder);
```

Comment:
it seems that we should either close dictionaryEncoder in the scope it is used or have a field referring to it in in the top class to reuse it.